### PR TITLE
Created new collaborations list page and collaboration creation

### PR DESCRIPTION
**Pull request by BJSS-RicardoTavares**
Changed 'manage-innovation' path to 'manage'
Renamed mange info to manage-overview
Added collaborators to innovation overview
Created collaborations list page
Created collaborations creation wizard

Still WIP, as several visual improvements are needed.

image

image

image

image

### CODE REVIEW

File: src/modules/feature-modules/innovator/pages/innovation/manage/manage-collaborators-wizard.component.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+  }
+
+  onSubmitWizard(): void {
+
+    // if (!this.parseForm()) { return; }
+
+    // if (!this.form.valid) { return; }
+
+    console.log(this.wizard.currentAnswers);
```
Comment (manuelsousa-bjss) [inline src/modules/feature-modules/innovator/pages/innovation/manage/manage-collaborators-wizard.component.ts:123]: remeber to remove this on the next PR ;)

File: src/modules/feature-modules/innovator/pages/innovation/overview/overview.component.html [hunk 2]
```diff
@@ -10,7 +10,7 @@ <h2 class="nhsuk-card--care__heading">
       </div>
 
       <div class="nhsuk-card__content">
-        <p> There are no organisation units currently supporting your innovation. There are also no more outstanding suggestions from your original needs assessment or from other organisations. </p>
+        <p> There are no organisation units currently supporting your innovation?. There are also no more outstanding suggestions from your original needs assessment or from other organisations. </p>
```
Comment (manuelsousa-bjss) [inline src/modules/feature-modules/innovator/pages/innovation/overview/overview.component.html:13]: imo that ? shouldn't be there

File: src/modules/shared/services/innovations.service.ts [hunk 3]
```diff
@@ -353,10 +407,10 @@ export class InnovationsService extends CoreService {
       }))
     );
   }
-  
-  getInnovationSubmission(innovationId: string): Observable<InnovationSubmissionDTO> {
+
+  getInnovationSubmission(innovationId: string): Observable<{ submittedAllSections: boolean, submittedForNeedsAssessment: boolean }> {
     const url = new UrlModel(this.API_INNOVATIONS_URL).addPath('v1/:innovationId/submissions').setPathParams({ innovationId });
-    return this.http.get<InnovationSubmissionDTO>(url.buildUrl()).pipe(take(1), map(response => response));
+    return this.http.get<{ submittedAllSections: boolean, submittedForNeedsAssessment: boolean }>(url.buildUrl()).pipe(take(1), map(response => response));
```
Comment (manuelsousa-bjss) [inline src/modules/shared/services/innovations.service.ts:413]: Didn't comment before when I saw removing and changing types. But I prefer having DTOs for API responses than this sort of types.

Then the method either uses the DTO if it doesn't make any changes or it has a different return type when it does do changes.

This approach has two issues in my opinion:
- it doesn't have one place to go when changing a DTO from the API
- it doesn't ensure that response only has these two fields
specially when they are the same at

File: src/modules/feature-modules/innovator/pages/innovation/manage/manage-collaborators-overview.component.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    ]).subscribe(([activeCollaborators, historyCollaborators]) => {
+
+      let action: null | { label: string, url: string } = null;
+
+      this.activeCollaborators.setData(activeCollaborators.data.map(item => {
+
+        switch (item.status) {
+
+          case InnovationCollaboratorStatusEnum.ACTIVE:
+            action = { label: 'Manage', url: 'info' };
```
Comment (diogo-costa-bjss) [inline src/modules/feature-modules/innovator/pages/innovation/manage/manage-collaborators-overview.component.ts:67]: Every action will go to the same place (`/:collaboratorId`), right?

**Review by manuelsousa-bjss — APPROVED** [review]

**Pull request merged**